notebooks/thunder_as_torch_compile_backend_tutorial.ipynb

```diff
@@ -0,0 +1,308 @@
+{
```

Comment:
<p>Before getting into technical details, if we follow the proposed structure so far we've introduced ThunderFX as a thing that can optimize PyTorch CUDA programs, we've seen how to pass a simple PyTorch program to it, and we've learned that ThunderFX supports both CPU and CUDA tensors.</p><p> </p><p>What is the next most important thing for a practitioner to understand? I think an example of Thunder running a more realistic PyTorch program, maybe a module from a popular deep learning model, and showing how it improves performance over PyTorch eager might be interesting?</p><p> </p><p>Then we'd be saying:</p><p> </p><ol><li>ThunderFX can optimize PyTorch CUDA programs</li><li>You can use it like this</li><li>It supports CPU and CUDA tensors, but current optimizations target CUDA</li><li>Here's a more realistic example using CUDA where you can see it runs faster</li></ol><p> </p><p>I think those are the most interesting things to say immediately.</p><p> </p><p>I'm going to stop reviewing here for now to see how you like these suggestions, @kiya00. Let me know if you'd like to chat more over a VC, too. The tutorial could keep going to cover some of the technical internals, introspection, executor selection, etc., but I think stopping after (4) may be a good "Hello, World!" Maybe this notebook should then direct the reader to another notebook about the internals and introspection?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Lightning-AI/lightning-thunder/pull/1524/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Reply:
Hi @mruberry , thank you for the suggestion!! I'm actually not sure how to write the tutorial, so in this PR the structure is: 1. introduces that besides using `thunder.jit` on the function directly, Thunder can also work as a backend of torch.compile (including some basic intro/links to what torch.compile is) 2. gives an example of using thunderFX interface 3. gives some inspect on how thunderFX handles the FX graph, and introduces the interfaces users can use to inspect the FX graph that thunder accelerated(for user to debug).
I agree that as an "hello world" we should give more basic examples like with cpu/cuda tensors, and also the motivation of using thunderFX -- how the performance is improved. I'll modify the structure accordingly.
And for the UX part, I create a new issue and comment there so we can let more people see it  https://github.com/Lightning-AI/lightning-thunder/issues/1529

Reply:
Sounds great! Let's make the UX changes and I look forward to reading the next draft!